internal/api/api_client_test.go

```diff
@@ -0,0 +1,53 @@
+package api
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type MockClient struct {
+	DoFunc func(req *http.Request) (*http.Response, error)
```

Comment:
this is unused, I'd prefer this over the global `GetDoFunc` below

Reply:
I'd also recommend looking at mockery for generating these. You can take a look at https://github.com/ministryofjustice/opg-data-lpa-store/pull/253 for how that can be done

Reply:
done